### PR TITLE
Sprint 15

### PR DESCRIPTION
**Pull request by ericsorenson**
# Sprint 15

## 1652 Upgrade to USWDS 2.0: File a Petition
image

## 1658 New Design: All Messages/QC Review
image

## 1659 New Design: Case Detail Page
image

## 1660 New Design: Document Detail Page
image

## 1661 New Design: File an External Document Flow
image

## 1664 New Design: Start a Case Form
image

## 1662 New Design: File a Petition - “Before You Begin” Page
image

## 1663 New Design: External User Dashboard and Error Pages
image

## 1187 Case gets set as “General Docket - At Issue (Ready for Trial)”
Case status is automatically set to "General Docket - At Issue (Ready for Trial)" at the time that all of these conditions become present:
1) An answer-type document has been filed in the case [see note]
2) 45 days have passed after the filing of an answer-type document
3) Case is of status 'General Docket - Not At Issue'

### CODE REVIEW

**Comment by waldoj** [comment]
Here's the review checklist that I'll work my way through, in reviewing this PR:

- [ ] begin by deleting any existing copy of the Git repository and cloning a fresh copy, running `docker system prune` to remove any cached artifacts from any prior code review, and running `./clean-up.sh` to update Node packages
- [ ] review the pull request itself, to get oriented
	- [ ] read the description of the pull request, which should summarize the changes made
	- [ ] read through every task on the Scrum board that's encompassed by this pull request
	- [ ] read the description of the commits that comprise the pull request
- [ ] stand up the site locally
	- [ ] test all functionality in all major browsers, emphasizing the functionality that this pull request addresses
		- [ ] for internal Court functionality, perform the most thorough testing in Chrome, though also test in Edge and Firefox
		- [ ] for public-facing functionality, test in browsers consistent with [public browser use data](https://analytics.usa.gov/), including Internet Explorer 11
		- [ ] test in Mobile Safari and Mobile Chrome, with the caveat that not all internal Court functionality will be necessary on these platforms
	- [ ] use an automated audit tool for code quality and practices (recommended: [Chrome DevTools](https://developers.google.com/web/tools/chrome-devtools/), aka [Lighthouse](https://chrome.google.com/webstore/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk))
		- [ ] look at efficiency of page loads, asset sizes, HTTP connection management, etc.
	- [ ] review for accessibility
		- [ ] use an automated audit tool, such as Chrome Audit or aXe
		- [ ] navigate site only with the keyboard
		- [ ] use VoiceOver or Narrator to navigate the site with audio only, with the display turned off
		- [ ] manually test anything that pa11y cannot test automatically (e.g., contrast of text over images)
- [ ] review static code analysis results in SonarCloud
- [ ] run a security audit of dependencies (`npm audit`) in `efcms-service`, `web-client`, and `shared`, to ensure that there are no vulnerabilities that will be deployed to production (as opposed to vulnerabilities that only have an impact on the development environment)
- [ ] examine OWASP ZAP output in `docs/`, to ensure that any errors are known to be false positives or have been previously declared to be acceptable
- [ ] skim all new code, in the context of existing code, [looking for problems](https://github.com/ustaxcourt/ef-cms/blob/staging/docs/CODE_REVIEW.md#what-we-look-for) (knowing that the vast majority of new code will be covered by tests)
- [ ] review all tests
	- [ ] look at code coverage of tests in SonarCloud
	- [ ] methodically review all new tests for correctness, quality of naming
- [ ] determine what code isn’t tested, review that rigorously
- [ ] review documentation to ensure that it matches changes
- [ ] provide comments on the pull request on GitHub, as necessary
	- [ ] for comments that are specific to a particular line of code, comment on those specific lines
	- [ ] for comments that are more general, attach the comment to a random line in `README.md` (as opposed to commenting on the pull request itself), to be able to use GitHub's ability to thread discussions on those comments
- [ ] for each feature-level bug (i.e., it’s working as designed, but designed wrong), open a new issue and put it in the backlog

**Comment by waldoj** [comment]
Because GitHub lacks effective pull-request-level feedback mechanisms that must be addressed before a pull request can be merged, I'll be using `README.md` as the place to attach PR-level comments.

File: README.md [hunk 1]
```diff
@@ -4,15 +4,15 @@
...
...oud.io/api/project_badges/measure?project=ef-cms-front-end&metric=sqale_rating)](https://sonarcloud.io/dashboard?id=ef-cms-front-end)<br>[![Security Rating](https://sonarcloud.io/api/project_badges/measure?project=ef-cms-front-end&metric=security_rating)](https://sonarcloud.io/dashboard?id=ef-cms-front-end)<br>[![Known Vulnerabilities](https://snyk.io/test/github/flexion/ef-cms/badge.svg?targetFile=web-client%2Fpackage.json)](https://snyk.io/test/github/flexion/ef-cms?targetFile=web-client%2Fpackage.json) | [![Coverage](https://sonarcloud.io/api/project_badges/measure?project=ef-cms-shared&metric=coverage)](https://sonarcloud.io/dashboard?id=ef-cms-shared)<br>[![Maintainability Rating](https://sonarcloud.io/api/project_badges/measure?project=ef-cms-shared&metric=sqale_rating)](https://sonarcloud.io/dashboard?id=ef-cms-shared)<br>[![Security Rating](https://sonarcloud.io/api/project_badges/measure?project=ef-cms-shared&metric=security_rating)](https://sonarcloud.io/dashboard?id=ef-cms-shared)<br>[![Known Vulnerabilities](https://snyk.io/test/github/flexion/ef-cms/badge.svg?targetFile=shared%2Fpackage.json)](https://snyk.io/test/github/flexion/ef-cms?targetFile=shared%2Fpackage.json)
```
[the remark this answers is not in the document]
Comment (ericsorenson) [reply README.md]: Were you using Docker or running directly on your computer?

[the remark this answers is not in the document]
Comment (waldoj) [reply README.md]: Using Docker.

File: shared/src/business/utilities/DateHandler.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ * @param {string} dateString a date string to be sent to persistence
+ * @param {string} inputFormat
+ * @returns {string} a formatted ISO date string
+ */
+module.exports.createISODateString = (dateString, inputFormat) => {
+  let result;
+  if (!dateString) {
+    result = moment();
+  } else {
+    result = prepareDateFromString(dateString, inputFormat, USTC_TZ);
```
[the remark this answers is not in the document]
Comment (ericsorenson) [reply shared/src/business/utilities/DateHandler.js]: Updated!

File: web-client/src/presenter/computeds/formattedCaseDetail.js [hunk 3]
```diff
@@ -68,7 +75,11 @@ const formatYearAmount = (caseDetailErrors, caseDetail) => (
   };
 };
 
-export const formatYearAmounts = (caseDetail, caseDetailErrors = {}) => {
+export const formatYearAmounts = (
+  caseDetail,
+  caseDetailErrors = {},
```
[the remark this answers is not in the document]
Comment (ericsorenson) [reply web-client/src/presenter/computeds/formattedCaseDetail.js]: Updated!

File: web-client/src/presenter/computeds/formattedCaseDetail.js [hunk 4]
```diff
@@ -87,6 +98,7 @@ const formatDocketRecordWithDocument = (
   caseDetail,
   docketRecords = [],
   documents = [],
+  applicationContext,
```
[the remark this answers is not in the document]
Comment (ericsorenson) [reply web-client/src/presenter/computeds/formattedCaseDetail.js]: Updated!

File: README.md [hunk 5]
```diff
@@ -76,7 +76,7 @@ The EF-CMS is comprised of two components: the API and the UI. Both must be run
 
 ### Prerequisites
 
-- Node v8.10.0
+- Node v10.15.3
```
[the remark this answers is not in the document]
Comment (ericsorenson) [reply README.md]: Updated, a table cell now uses the default pointer.

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ericsorenson) [reply README.md]: That's the hot reload websockets connection from Parcel. It uses a random port, so we can't specifically open it for Docker. Nothing is broken, it just won't automatically reload the browser on change.

[the remark this answers is not in the document]
Comment (waldoj) [reply README.md]: Interesting! And, yes, quite harmless — I'll just ignore this.

File: web-client/src/presenter/actions/FileDocument/clearWizardDataAction.test.js [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+          },
+        },
+      },
+    });
+
+    expect(result.state.form.secondaryDocument).toEqual({
+      category: 'what',
+    });
+  });
+  it('clears document scenario with "secondaryDocument.documentType" key', async () => {
```
[the remark this answers is not in the document]
Comment (ericsorenson) [reply web-client/src/presenter/actions/FileDocument/clearWizardDataAction.test.js]: Updated!

File: web-client/src/presenter/actions/FileDocument/clearWizardDataAction.test.js [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+          secondarySupportingDocumentFreeText: 'test',
+        },
+      },
+    });
+
+    expect(result.state.form).toEqual({
+      secondarySupportingDocumentFreeText: null,
+    });
+  });
+  it('clears document scenario with "secondaryDocument.category" key', async () => {
```
[the remark this answers is not in the document]
Comment (ericsorenson) [reply web-client/src/presenter/actions/FileDocument/clearWizardDataAction.test.js]: Updated!

File: shared/src/business/useCases/checkForReadyForTrialCasesInteractor.test.js [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+      });
+    } catch (e) {
+      error = e;
+    }
+
+    expect(error).toBeUndefined();
+    expect(updateCaseSpy.called).toEqual(false);
+  });
+
+  it("should update cases to 'ready for trial' that meet requirements", async () => {
```
[the remark this answers is not in the document]
Comment (ericsorenson) [reply shared/src/business/useCases/checkForReadyForTrialCasesInteractor.test.js]: @wbyoko would you please add some comments here?

[the remark this answers is not in the document]
Comment (wbyoko) [reply shared/src/business/useCases/checkForReadyForTrialCasesInteractor.test.js]: Requirements have been added.

[the remark this answers is not in the document]
Comment (ericsorenson) [reply shared/src/business/useCases/checkForReadyForTrialCasesInteractor.test.js]: Updated!

**Pull request merged**